### ISSUE TITLE
T14s Gen 4 AMD only boots when grub2 is selected. NO ISO boots in default mode!

### ISSUE THREAD

**Issue by spxak1** [ventoy/Ventoy#2729]
### Official FAQ

- [X] I have checked the official FAQ.

### Ventoy Version

1.0.97

### What about latest release

Yes. I have tried the latest release, but the bug still exist.

### Try alternative boot mode

Yes. I have tried them, but the bug still exist.

### BIOS Mode

UEFI Mode

### Partition Style

GPT

### Disk Capacity

64GB

### Image file checksum (if applicable)

Yes.

### What happened?

Tried and tested USB stick, works with every other laptop/computer. I have since tried different USB stick, fresh ventoy install, with/without secureboot support, fresh ISOs etc. The behaviour is **always** the same with **all** ISOs.

If I select default mode, the LED of the USB stick blinks for 3 times for 1 second and the system freezes (hard lock) with a black screen and a cursor at the top left. Needs power down/restart.

If an ISO offers ```grub 2 mode``` it boots fine. For Windows the ```winmi``` mode (spelling?) works. 

This laptop is based around the AMD 7840u, has a ton of security options in the bios, I have tested every single one of them, one at a time, the behaviour is the same. **Secure boot is off**.

These are the ISO's currently on this Ventoy stick:

~~~
 archlinux-2024.01.01-x86_64.iso
 debian-live-12.4.0-amd64-gnome.iso
 Endeavouros-Galileo-11-2023.iso
 Fedora-Everything-netinst-x86_64-39-1.5.iso
 Fedora-Workstation-Live-x86_64-39-1.5.iso
 FreeBSD-13.2-RELEASE-amd64-dvd1.iso
 garuda-gnome-linux-zen-231029.iso
 manjaro-gnome-23.1.3-240113-linux66.iso
 Nobara-39-GNOME-2024-01-11.iso
 pop-os_22.04_amd64_nvidia_37.iso
 proxmox-ve_8.0-2.iso
 tiny10 x64 23h2.iso
 tiny11 23H2 x64.iso
 ubuntu-22.04.3-desktop-amd64.iso
 ubuntu-23.10.1-desktop-amd64.iso
 Win10_22H2_EnglishInternational_x64.iso
 Win11_23H2_EnglishInternational_x64.iso
~~~

NONE of these boot in normal mode. The ALL boot in grub2 mode and winmi for the Windows ISO's.

A similar issue is reported here with another AMD based ThinkPad of the same generation, although the OP only tested clonezilla, with the same exact result: [https://forums.ventoy.net/showthread.php?tid=2765](https://forums.ventoy.net/showthread.php?tid=2765)

**Comment by munawar77** [comment]
Hi did you come across a solution or tried other boot makers instead? I own T14 Gen 3 and had exact same issues with Ventoy when selecting grub 2 mode. After managed to install openSUSE Tumbleweed, my system badly froze into blank screen with again stuck cursor on top left. Frustration indeed

**Comment by steve6375** [comment]
Does easy2boot.xyz work?

**Comment by spxak1** [comment]
> Hi did you come across a solution or tried other boot makers instead? I own T14 Gen 3 and had exact same issues with Ventoy when selecting grub 2 mode. After managed to install openSUSE Tumbleweed, my system badly froze into blank screen with again stuck cursor on top left. Frustration indeed

@munawar77 Grub 2 works. I also have OpenSuse TW and it works fine, so not sure what is wrong there.

**Comment by metal450** [comment]
Following to hopefully be notified if this gets fixed.

**Comment by metal450** [comment]
Seems to be related: any .img I try to boot that was created with ventoy/espack also behave the same. They don't offer a wimboot/grub2 option, so I cannot find any way to boot them. This is a pretty bad issue, as it means it's impossible to boot things like VeraCrypt Rescue...

**Comment by rgerardi** [comment]
Having the same issue with a Lenovo T14 Gen 5 AMD. No ISOs can boot in normal mode, just grub2 mode. Tested the same USB / Ventoy on a T14 Gen 2 and it booted the ISOs with no issues. Any ideas here?

**Comment by steve6375** [comment]
Did you try latest version of Ventoy?
Did you try making usb drive with mbr partitions instead of gpt?
Is there any bios update for the computer?
Did you report this as a bug to the manufacturer of the pc since other computers work ok?
It's hard for anyone who hasn't got this particular computer to reproduce the problem, so the more info you can provide might give the developer a clue as to where to look for the problem and work around it.

**Comment by Ricent82** [comment]
>Did you try latest version of Ventoy?

Yes, have updated multiple times since discovering this issue; just tried again with the latest 1.1.05 to be sure, & the behavior is the same.

> Did you try making usb drive with mbr partitions instead of gpt?

Yes, have tried as both MBR and GPT. The behavior is the same.

> Is there any bios update for the computer?

No, the BIOS is up to date.

> It's hard for anyone who hasn't got this particular computer to reproduce the problem

Fair enough, however it does sound like it's prevalent among recent ThinkPads - above reports mention at least the following models:
* T14s Gen 4 AMD
* T14 Gen 3
* T14 Gen 5 AMD
* And I should add that mine is P14s Gen 5 AMD

Given that ThinkPad is one of the top-selling laptop lines (and according to Claude, T14 is the best-selling of those), it does sound like this is more widespread than just one rare/fringe laptop.

> The more info you can provide might give the developer a clue as to where to look for the problem and work around it.

Would be happy to provide any info I can, but I'm not really sure what else that'd be beyond the behavior I'm observing?

> Did you report this as a bug to the manufacturer of the pc since other computers work ok?

Not really sure what I could report to Lenovo? They're a huge company, imagine saying "There's this open-source project on github that makes it easy to boot ISOs from USB, and it doesn't work on your laptops." I can't imagine their answer would be anything except "...We have tens of millions of laptops in the field, if you're having issue with one piece of software, you should try to contact its developer"

**Comment by steve6375** [comment]
Ventoy is used by millions of people and afaik this issue only affects the models you listed.
Manufacturers tend to have in-house BIOS development teams and the UEFI firmware is probably the same across their range except for hardware driver differences to cater for different chip peripheral etc 
Ventoy works on VMware and virtual box as well as 99pc of other UEFI systems, so there is something different about these particular models which could be due to the firmware or possibly the memory map or other hardware differences due to the chipset used?
What is different about these systems?
Do these systems have thunderbolt? Have you tried all ports? Can thunderbolt be disabled in the bios settings? Anything else different about these systems?

**Comment by Ricent82** [comment]
> Do these systems have thunderbolt?

My USB key is USB-A - I don't think thunderbolt is applicable to USB-A?

> Have you tried all ports?

I've tried all USB-A ports, yes

> Anything else different about these systems?

Honestly your guess is better than mine - I'm not knowledgeable enough about what would be relevant to say. Like...sure, there are lots of differences: Different from Intel laptops, they use AMD; Different from Dells, they use Lenovo BIOS, etc...?

> Ventoy is used by millions of people and afaik this issue only affects the models you listed.

I mean yes, there's some compatibility issue between Ventoy & recent ThinkPads.

Like when your computer crashes, Microsoft could "Windows works on all other laptops, so it must be the hardware."
But then you boot a live Linux distro & say "it runs Linux just fine, so it must be Windows."
The truth is neither, it's compatibility between the two.

So either Ventoy needs some code update to support newer Lenovo systems, or Lenovo has to rework their BIOS to make sure Ventoy works. But honestly the chance of the latter happening is pretty much 0.

**Comment by leo-b** [comment]
I am experiencing the same issues on my Thinkpad T14 Gen 5 (AMD).
Ventoy normal and memdisk boot both hang showing a black screen with a solid (non-blinking) cursor.

> Does [www.easy2boot.xyz](http://www.easy2boot.xyz/) work?

Yes. easy2boot works using both `partnew/Easy2Boot [.isodef]` and `mem [.isomem]` modes.

**Comment by tokit3mx** [comment]
any news on this? facing the same issue on thinkpad p14s g5 with ryzen 7 8840hs

**Comment by pthivent** [comment]
Same issue with a ThinkPad T14 Gen 4 (AMD), Ryzen 7840U, with up to date BIOS and firmware.

> BIOS version: R2FET68W (1.48 )

**Comment by My1** [comment]
gonna hop in this issue too. Thinkpad L16 Gen2 AMD (Ryzen 250; Model Type 21SC)

**Comment by cuter-doc0** [comment]
I can confirm this is an issue with THINKPAD L16 AMD GEN 2 - Ryzen 250
